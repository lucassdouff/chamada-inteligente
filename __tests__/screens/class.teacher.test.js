import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContext } from '@react-navigation/native';
import ClassScreen from '../../src/app/screens/teacher/Class/Class';
import { Location } from 'expo';

const navContext = {
    isFocused: () => true,
    addListener: jest.fn(() => jest.fn())
}

jest.mock('expo', () => ({
    Location: {
      requestForegroundPermissionsAsync: jest.fn(),
      getCurrentPositionAsync: jest.fn(),
    },
}));

describe('Class screen should render correctly', () => {

    afterEach(() => {
        jest.resetAllMocks();
    });

    it('student class renders correctly', async () => {

        Location.requestForegroundPermissionsAsync.mockResolvedValueOnce({ status: 'granted' });
        
        const mockedParams = {
            route: { params: { userClass: {
                id_class: 1,
                name: 'Class 1',
                code: 123456,
                semester: '2021-1',
                id_teacher: 1,
                id_course: 1,
                class_weekdays: [{
                    weekday: 1,
                    start_hour: '08:00:00',
                    end_hour: '10:00:00'
                }]
            }}
            },
            navigation: ''
        };

        await expect(render(
            <NavigationContext.Provider value={navContext}>
                <ClassScreen {...mockedParams} />
            </NavigationContext.Provider>
        )).toBeTruthy();
    });
    
    afterAll(async () => {
        await new Promise(resolve => setTimeout(() => resolve(), 100)); // avoid jest open handle error
    });
});
