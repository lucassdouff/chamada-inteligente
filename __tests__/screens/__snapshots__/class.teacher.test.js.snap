// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Class screen should render correctly teacher class renders correctly 1`] = `
<RCTScrollView
  collapsable={false}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  style={
    [
      {
        "flexDirection": "column",
      },
      {
        "paddingBottom": 8,
        "paddingTop": 8,
      },
      {
        "paddingLeft": 16,
        "paddingRight": 16,
      },
      {
        "width": "100%",
      },
      {
        "marginTop": 8,
      },
    ]
  }
  waitFor={
    [
      {
        "current": null,
      },
    ]
  }
>
  <View>
    <View
      style={
        [
          {
            "marginBottom": 24,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "height": 96,
            },
            {
              "display": "flex",
            },
            {
              "flexDirection": "column",
            },
            {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            {
              "paddingBottom": 8,
              "paddingTop": 8,
            },
            {
              "borderBottomWidth": 1,
              "borderLeftWidth": 1,
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            {
              "borderBottomColor": "#9ca3af",
              "borderLeftColor": "#9ca3af",
              "borderRightColor": "#9ca3af",
              "borderTopColor": "#9ca3af",
            },
            {
              "borderBottomLeftRadius": 4,
              "borderBottomRightRadius": 4,
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
            {
              "marginBottom": 12,
            },
            {
              "backgroundColor": "#2563eb",
            },
          ]
        }
        testID="class-card"
      >
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              {
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "width": "80%",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                  },
                  {
                    "fontSize": 18,
                    "lineHeight": 28,
                  },
                ]
              }
            >
              123456 - Class 1
            </Text>
          </View>
          <View
            style={
              {
                "overflow": "hidden",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
        <Text
          style={
            [
              {
                "color": "#fff",
              },
              {
                "marginBottom": 8,
              },
            ]
          }
        >
          2021-1
        </Text>
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#fff",
                },
              ]
            }
          >
            Segunda - 08:00 às 10:00
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "alignSelf": "center",
            },
            {
              "width": "75%",
            },
            {
              "marginTop": 16,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "marginBottom": 8,
              },
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {},
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    {
                      "color": "blue",
                    },
                  ]
                }
              >
                INICIAR CHAMADA
              </Text>
            </View>
          </View>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              [
                {},
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  {
                    "color": "orange",
                  },
                ]
              }
            >
              AGENDAR CHAMADA
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "marginBottom": 16,
          },
          [
            {
              "borderBottomColor": "#6b7280",
              "borderLeftColor": "#6b7280",
              "borderRightColor": "#6b7280",
              "borderTopColor": "#6b7280",
            },
            {
              "borderBottomWidth": 0,
              "borderTopWidth": 1,
            },
          ],
        ]
      }
    >
      <Text
        style={
          [
            {
              "marginBottom": 16,
              "marginTop": 16,
            },
            {
              "fontSize": 20,
              "lineHeight": 28,
            },
          ]
        }
      >
        Informações da Turma
      </Text>
      <View
        style={
          [
            {
              "flexDirection": "column",
            },
            {
              "marginLeft": -16,
              "marginTop": -16,
            },
            {
              "paddingBottom": 8,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 8,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              {
                "justifyContent": "space-between",
              },
              [
                {
                  "marginLeft": 16,
                  "marginTop": 16,
                },
              ],
            ]
          }
        >
          <Text>
            Total de alunos inscritos:
          </Text>
          <Text>
             aluno(s)
          </Text>
        </View>
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              {
                "justifyContent": "space-between",
              },
              [
                {
                  "marginLeft": 16,
                  "marginTop": 16,
                },
              ],
            ]
          }
        >
          <Text>
            Média de frequência dos alunos:
          </Text>
          <Text>
             aluno(s)
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "marginBottom": 16,
          },
          [
            {
              "borderBottomColor": "#6b7280",
              "borderLeftColor": "#6b7280",
              "borderRightColor": "#6b7280",
              "borderTopColor": "#6b7280",
            },
            {
              "borderBottomWidth": 0,
              "borderTopWidth": 1,
            },
          ],
        ]
      }
    >
      <Text
        style={
          [
            {
              "marginBottom": 16,
              "marginTop": 16,
            },
            {
              "fontSize": 20,
              "lineHeight": 28,
            },
          ]
        }
      >
        Histórico de chamadas
      </Text>
      <View
        style={
          [
            {
              "alignSelf": "center",
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "width": "100%",
              },
            ]
          }
        >
          <RCTScrollView
            horizontal={true}
          >
            <View>
              <View
                style={
                  [
                    {
                      "flexDirection": "column",
                    },
                    {
                      "borderBottomWidth": 1,
                      "borderLeftWidth": 1,
                      "borderRightWidth": 1,
                      "borderTopWidth": 1,
                    },
                  ]
                }
              />
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      style={
        [
          {
            "borderBottomColor": "#6b7280",
            "borderLeftColor": "#6b7280",
            "borderRightColor": "#6b7280",
            "borderTopColor": "#6b7280",
          },
          {
            "borderBottomWidth": 0,
            "borderTopWidth": 1,
          },
        ]
      }
      visible={false}
    />
  </View>
</RCTScrollView>
`;
